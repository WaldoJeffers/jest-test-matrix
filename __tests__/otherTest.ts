import testMatrix from '../src';

interface CanPayOpts {
  amount: number;
  balance: number;
  status: 'active' | 'locked' | 'expired';
}

function canPay({amount, balance, status}: CanPayOpts){
  if (status === 'locked' || status == 'expired'){
    return false;
  }
  return balance > amount;
}

it('should only allow valid payments', () => {
  expect(testMatrix(canPay, {
    amount: [20, 100],
    balance : [50],
    status: ['active', 'locked', 'expired']
  })).toMatchSnapshot();
})