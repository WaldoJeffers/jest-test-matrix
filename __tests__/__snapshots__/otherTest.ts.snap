// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should only allow valid payments 1`] = `
.-----------------------------------.
|              canPay               |
|-----------------------------------|
| balance | amount | status  |      |
|---------|--------|---------|------|
|      50 |     20 | active  | true |
|         |        | locked  | false |
|         |        | expired | false |
|         |    100 | active  | false |
|         |        | locked  | false |
|         |        | expired | false |
'-----------------------------------'
`;
